apps/ewallet_config/lib/ewallet_config/setting.ex

```diff
@@ -0,0 +1,261 @@
+  def all do
+    StoredSetting
+    |> Query.order_by(asc: :position)
+    |> Repo.all()
+    |> Enum.map(&build/1)
+  end
+
+  @doc """
+  Retrieves a setting by its string name.
+  """
+  @spec get(String.t()) :: %Setting{}
+  def get(key) when is_atom(key) do
+    get(Atom.to_string(key))
+  end
+
+  def get(key) when is_binary(key) do
+    case Repo.get_by(StoredSetting, key: key) do
+      nil -> nil
+      stored_setting -> build(stored_setting)
+    end
+  end
+
+  def get(_), do: nil
+
+  @spec get(String.t()) :: any()
+  def get_value(key, default \\ nil)
+
+  def get_value(key, default) when is_binary(key) do
+    case Repo.get_by(StoredSetting, key: key) do
+      nil -> default
+      stored_setting -> extract_value(stored_setting)
+    end
+  end
+
+  def get_value(nil, _), do: nil
+
+  @doc """
+  Creates a new setting with the passed attributes.
+  """
+  @spec insert(Map.t()) :: {:ok, %Setting{}} | {:error, %Changeset{}}
+  def insert(attrs) do
+    attrs = cast_attrs(attrs)
+
+    %StoredSetting{}
+    |> StoredSetting.changeset(attrs)
+    |> Repo.insert()
+    |> return_from_change()
+  end
+
+  @doc """
+  Inserts all the default settings.
+  """
+  @spec insert_all_defaults(Map.t()) :: [{:ok, %Setting{}}] | [{:error, %Changeset{}}]
+  def insert_all_defaults(overrides \\ %{}) do
+    get_default_settings()
+    |> Enum.map(fn {key, data} ->
+      case overrides[key] do
+        nil ->
+          insert(data)
+
+        override ->
+          data
+          |> Map.put(:value, override)
+          |> insert()
+      end
+    end)
+    |> Enum.all?(fn res ->
+      case res do
+        {:ok, _} -> true
+        _ -> false
+      end
+    end)
+    |> return_insert_result()
+  end
+
+  defp return_insert_result(true), do: :ok
+  defp return_insert_result(false), do: :error
+
+  @spec update(String.t(), Map.t()) ::
+          {:ok, %Setting{}} | {:error, Atom.t()} | {:error, Changeset.t()}
+  def update(nil, _), do: {:error, :setting_not_found}
+
+  def update(key, attrs) when is_binary(key) and is_map(attrs) do
+    case Repo.get_by(StoredSetting, %{key: key}) do
+      nil ->
+        {:error, :setting_not_found}
+
+      setting ->
+        attrs = cast_attrs(attrs)
+
+        setting
+        |> StoredSetting.update_changeset(attrs)
+        |> Repo.update()
+        |> return_from_change()
+    end
+  end
+
+  @spec update_all(List.t()) :: [{:ok, %Setting{}} | {:error, Atom.t()} | {:error, Changeset.t()}]
+  def update_all(attrs_list) when is_list(attrs_list) do
+    Enum.map(attrs_list, fn attrs ->
+      update(attrs["key"] || attrs[:key], attrs)
+    end)
+  end
+
+  defp return_from_change({:ok, stored_setting}) do
+    {:ok, build(stored_setting)}
+  end
+
+  defp return_from_change({:error, changeset}) do
+    {:error, build_changeset(changeset)}
+  end
+
+  defp cast_attrs(attrs) do
+    attrs
+    |> cast_value()
+    |> cast_options()
+    |> add_position()
+  end
+
+  defp build_changeset(changeset) do
```

Comment:
Because that first changeset is tainted, it contains details about the internal schema `StoredSetting` so we need to replace/clean the returned changeset before sending it back to the sender.